lightning/src/ln/outbound_payment.rs

```diff
@@ -473,6 +473,18 @@ impl RecipientOnionFields {
 	}
 }
 
+/// Arguments for [`super::channelmanager::ChannelManager::send_payment_along_path`].
+pub(super) struct SendAlongPathArgs<'a> {
```

Comment:
nit: Not so sure about the location of this struct. i'd think its supposed to be more co-located with where its used but i also think `channelmanager.rs` is also too large. 

what is the current understanding around `outbound_payment.rs`?  are we intending to move most/all outbound payment / routing / orchestration logic here?

Reply:
The `outbound_payment` module came about so we could remove code from `channelmanager`. So, anything exclusive to outbound payments (such as this struct) could reasonably live there IMO. I believe we've moved most if not all of it already.